governance/src/lib.rs

```diff
@@ -29,15 +29,33 @@ pub struct Governance<N: GossipNetwork, S: Storage> {
 
 impl<N: GossipNetwork, S: Storage> Governance<N, S> {
     pub async fn create(_dms: DMS<N, S>, _height: BlockHeight) -> Result<(), Error> {
-        unimplemented!()
+        Ok(())
     }
 
-    pub async fn open(_dms: DMS<N, S>) -> Result<Self, Error> {
-        unimplemented!()
+    pub async fn open(dms: DMS<N, S>) -> Result<Self, Error> {
+        Ok(Self { dms })
     }
 
-    pub async fn read(&self) -> Result<GovernanceState, Error> {
-        unimplemented!()
+    pub async fn read(&self) -> Result<GovernanceStatus, Error> {
+        let messages = self.dms.read_messages().await?;
+        let votes = messages
+            .iter()
+            .map(|message| {
+                (
+                    message.data().to_string().to_hash256(),
+                    message.signature().signer().clone(),
+                )
+            })
+            .fold(HashMap::new(), |mut votes, (data_hash, voter)| {
+                votes
+                    .entry(data_hash)
+                    .or_insert_with(HashSet::new)
+                    .insert(voter);
+                votes
+            });
+        let height = self.dms.read_height().await?;
+        let status = GovernanceStatus { votes, height };
+        Ok(status)
     }
 
     pub async fn vote(
```

Comment:
I can't accept the function name `vote` here . 
This function makes a data, makes a message from data and adds to storage and broadcast. 
Can we think about a better function name? (i.e. add_message_and_broadcast or something mean like that)